service-base/src/main/java/org/eclipse/hono/service/registration/BaseRegistrationService.java

```diff
@@ -182,15 +182,27 @@ public final void processRegistrationMessage(final Message<JsonObject> regMsg) {
                 removeDevice(tenantId, deviceId, result -> reply(regMsg, result));
                 break;
             default:
-                log.info("operation [{}] not supported", operation);
-                reply(regMsg, RegistrationResult.from(HttpURLConnection.HTTP_BAD_REQUEST));
+                processOtherRegistrationMessage(regMsg, operation);
+                break;
             }
         } catch (ClassCastException e) {
             log.debug("malformed request message");
             reply(regMsg, RegistrationResult.from(HttpURLConnection.HTTP_BAD_REQUEST));
         }
     }
 
+    /**
+     * Override the following method to extend RegistrationService with implementation
+     * specific operations.
+     *
+     * @param regMsg registration message to be processed
+     * @param operation implementation specific operation to be executed.
+     */
+    public void processOtherRegistrationMessage(final Message<JsonObject> regMsg, String operation) {
```

Comment:
*processCustomRegistrationMessage* and declare *protected*

Reply:
will change it to `processCustomRegistrationMessage`